### DISCUSSION TITLE
Tauri Platform

### DISCUSSION THREAD

**Discussion by macchie** [ionic-team/capacitor#4321] [Ideas]
Hello everyone,
i was wondering if there is any plan to implement the [Tauri Project](https://tauri.studio/en/) as a Capacitor platform.
Electron is good and quite portable but apps get really big, Tauri looks a really promising project.

Could i technically fork the Electron platform and build my own integration?

Thanks for your work!

**Comment by jcesarmobile** [comment]
Yeah, in Capacitor 2.3 we added support for 3rd party platforms and made electron a community maintained platform, so you can fork @capacitor-community/electron and build your own.
Or start fresh taking it as a reference.
The important bits are those lines with the scripts that will be forwarded from Capacitor CLI to your own scripts that run add, copy, update, sync and open
https://github.com/capacitor-community/electron/blob/main/package.json#L11-L15

**Reply by macchie** [reply]
i'll give it a try! hope i'll produce something useseful :)

**Reply by darrylnoakes** [reply]
> i'll give it a try! hope i'll produce something useful :)

Did you make any progress with this?

I found Tauri today and am really excited about using it, but disappointed that I can't use it with Capacitor yet (afaik).
My primary project is built with Ionic and Capacitor, which makes mobile apps a snap, but as it is a relatively simple project (song viewer, like a really basic Ultimate Guitar), Electron is overkill.

**Reply by thomasvidas** [reply]
You can target the PWA build with Capacitor and then point that directly to as tauri web directory, you'll just miss out on some native APIs, but it is totally possible to build a native app with Tauri and Capacitor.

**Reply by darrylnoakes** [reply]
So I would build it as a web app, and then use Tauri with that build?
Can I use my normal web app build (not a PWA yet)? Or must it be a PWA?

I believed it was possible using this method, but was concerned that it would be nothing more than a fancy PWA, as I would not be able to use native APIs.
However, I think I will try it out, even if just to familiarize myself with Tauri.
Thanks for the confirmation.
Heck, I may end up using some Tauri-related code in my app for some of the stuff, if I really need to 😄.

One more question :).
The Capacitor Storage plugin falls back to `localStorage` when native APIs are not available.
What would `localStorage` be when using Tauri? A custom storage solution provided by it?

**Reply by thomasvidas** [reply]
It does not need to be a PWA, it can just be a web app. Sorry I was confusing! The plugins will all fall back to their web solutions if native API is not available; so yes the capacitor storage plugin would use `LocalStorage`

**Comment by IT-MikeS** [comment]
Been playing with this for a bit and may publish a Tauri community platform soon, lots it won't do out of the gate but for simple "containerized" type apps it will work well. 👍

**Reply by IT-MikeS** [reply]
Little update / preview for those following this discussion 🧐

![image](https://user-images.githubusercontent.com/20338451/128661969-ba6f2355-700b-45e6-aeb3-cace39fe4be3.png)

**Reply by darrylnoakes** [reply]
Looks good! 🎉

**Comment by IT-MikeS** [comment]
Well is still very much an alpha, but here is a Tauri platform for Capacitor 👍 (going to try and get `cap open..` to work on MacOS later tonight)

https://github.com/capacitor-community/tauri

**Comment by djabif** [comment]
Necroing this a bit but for anyone who lands here looking for the same thing: [Capawesome shipped a Tauri platform for Capacitor](https://github.com/capawesome-team/capacitor-tauri). 

Needs Capacitor 8+ and Tauri 2+. uses the system webview instead of bundling Chromium so installers come out around 3-10MB, and it auto generates deny-by-default capability files based on whatever plugins your app is using. workflow is the same as mobile too, cap add, cap sync, cap run.

Worth flagging the limits since they're basically what people were debating in this thread back in 2021: Tauri has no Node runtime, so random native Capacitor plugins just won't run, only a curated set (filesystem, preferences, dialog, local notifications) plus anything with a web fallback. Also no OTA updates for the web bundle since it gets compiled straight into the binary, so updates mean a new signed build.

If you need more plugins or actual live updates, you can use the [Electron platform](https://github.com/capawesome-team/capacitor-electron).
